File: misc/day1.py
with open("input01.txt") as file:
    lines = file.read().strip().split("\n")

zero_count = 0
click_count = 0
dial = 50
for line in lines:
    x = int(line[1:])
    if line[0] == 'L':
        x = -x
    x = dial + x
    next_dial = x % 100
    if next_dial == 0:
        zero_count += 1
    click_count += abs(x)//100
    if x < 0 and next_dial != 0:
        click_count += 1
    dial = next_dial

print("part one:", zero_count)
print("part two:", click_count)

File: misc/day1.ml
let read_entire_file path =
    let ch = open_in path in
    let s = really_input_string ch (in_channel_length ch) in
    close_in ch;
    s
let ( @% ) a b = (a mod b + b) mod b

let rec solve (dial, total) line =
    let turn = String.sub line 1 (String.length line - 1)
        |> int_of_string
        |> (if line.[0] = 'L' then (-) else (+)) dial in
    let clicks = (abs turn) / 100 + if turn < 0 && turn @% 100 != 0 then 1 else 0 in
    (turn @% 100, total + clicks)

let () =
    read_entire_file "input01.txt" |> String.trim |> String.split_on_char '\n'
    |> List.fold_left solve (50, 0) |> snd |> Printf.printf "%d\n"
